sdk/core/azure-core/test/ut/exception_test.cpp

```diff
@@ -33,7 +33,7 @@ TEST(RequestFailedException, JSONError)
   EXPECT_EQ(exception.RequestId, "1");
   EXPECT_EQ(exception.ClientRequestId, "2");
   EXPECT_EQ(exception.ReasonPhrase, "retry please :");
-  EXPECT_EQ(exception.what(), std::string("Received an HTTP unsuccessful status code."));
+  EXPECT_EQ(std::string(exception.what()).find("Received an HTTP unsuccessful status code"), 0);
```

Comment:
We're setting the error code and the ReasonPhrase, we can do exact match here instead of using find().

And we typically avoid this type of `expect_eq` usage, because if it is ever false, the test output would be `n is not equal to 0`, which is not helpful to understand why the test has failed.

Reply:
I didn't want to take too strong a dependency on the implementation of the error string - I thought the prefix was reasonable (especially given the previous tests) but wasn't sure about the rest of the string.

If we're not supposed to use `expect_eq`, what should we use instead of `expect_eq` here? Wouldn't the exact same issue apply to `expect_true`? If the error for `expect_eq` doesn't include the text being evaluated, which of the `expect` family includes the text?

Reply:
> I didn't want to take too strong a dependency on the implementation of the error string

Why not?  We need to make sure that we are adding the status code and any other thing we want to be there.  There should be a test which check this. Otherwise, we are not really testing the change you are adding in this PR.

> If we're not supposed to use expect_eq, what should we use instead of...

ah, ignore this.  I got confused.  @ahsonkhan , what was the recommended guideline here?   There was something we want to avoid due to the error output.  Do you remember?